admin/export/class-export-keywords-csv.php

```diff
@@ -0,0 +1,200 @@
+		}
+
+		return $csv;
+	}
+
+	/**
+	 * Formats a WPSEO_Export_Keywords_Query result as a CSV line.
+	 * In case of multiple keywords it will return multiple lines.
+	 *
+	 * @param array $result  A result as returned from WPSEO_Export_Keywords_Query::get_data.
+	 * @param array $columns The columns as returned from WPSEO_Export_Keywords_Query::get_columns.
+	 *
+	 * @return string A line of CSV, beginning with EOL.
+	 */
+	protected function format( $result, $columns ) {
+		// If our input is malformed return an empty string.
+		if ( ! is_array( $result ) || ! array_key_exists( 'ID', $result ) ) {
+			return '';
+		}
+
+		// Ensure we have arrays in the keywords.
+		$result['keywords'] = $this->get_array_from_result( $result, 'keywords' );
+		$result['keywords_score'] = $this->get_array_from_result( $result, 'keywords_score' );
+
+		$csv = '';
+
+		// Add at least one row plus additional ones if we have more keywords.
+		for ( $keywords_index = 0; $keywords_index < max( 1, count( $result['keywords'] ) ); $keywords_index++ ) {
+			$csv .= PHP_EOL . $this->sanitize_csv_column( $result['ID'] );
+
+			foreach ( $columns as $column ) {
+				if ( ! is_string( $column ) ) {
+					continue;
+				}
+				$csv .= $this->get_csv_column_from_result( $result, $column, $keywords_index );
+			}
+		}
+
+		return $csv;
+	}
+
+	/**
+	 * Returns a CSV column, including comma, from the result object based on the specified key
+	 *
+	 * @param array  $result         The result object.
+	 * @param string $key            The key of the value to get the CSV column for.
+	 * @param int    $keywords_index The number keyword to output.
+	 *
+	 * @return string
+	 */
+	protected function get_csv_column_from_result( $result, $key, $keywords_index ) {
+		switch ( $key ) {
+			case 'post_title':
+			case 'post_url':
+			case 'seo_score':
+				return $this->get_csv_string_column_from_result( $result, $key );
+			case 'keywords':
+			case 'keywords_score':
+				return $this->get_csv_array_column_from_result( $result, $key, $keywords_index );
+		}
+		return '';
+	}
+
+	/**
+	 * Returns an array from the result object.
+	 *
+	 * @param array  $result The result object.
+	 * @param string $key    The key of the array to retrieve.
+	 *
+	 * @return array
+	 */
+	protected function get_array_from_result( $result, $key ) {
+		if ( array_key_exists( $key , $result ) && is_array( $result[ $key ] ) ) {
+			return $result[ $key ];
+		}
+
+		return array();
+	}
+
+	/**
+	 * Returns a CSV column, including comma, from the result object by the specified key.
+	 * Expects the value to be a string.
+	 *
+	 * @param array  $result The result object to get the CSV column from.
+	 * @param string $key    The key of the value to get the CSV column for.
+	 *
+	 * @return string A CSV column including comma.
+	 */
+	protected function get_csv_string_column_from_result( $result, $key ) {
+		if ( is_array( $result ) && is_string( $key ) && array_key_exists( $key, $result ) ) {
```

Comment:
You know you are getting a `$key`, I think the `is_string` check is not necessary. You might consider typehinting the `$result`
`protected function get_csv_string_column_from_result( array $result, $key ) {`